### DISCUSSION TITLE
CodeAttribute class

### DISCUSSION THREAD

**Discussion by mariorancic01** [linq2db/linq2db#3768] [General]
I am playing with custom attributes and with the CodeAttribute class. 
There is lot of interfaces there but it is hard to guess concrete implementations.
Anyway, I want to add something that is not directly related to Linq2DB and database, but lets say that the same entity i might use in GRPC services. And then on the class I would like to put attribute [DataContract],
on the property i would like to put custom attribute [DataMember(Order = 1)]
Is it possible to do it with the CodeAttribute class

**Comment by MaceWindu** [comment]
Yes, you can use it for any attributes, not only for linq2db attributes

**Comment by croban** [comment]
### My Problem 
When attempting to add attributes, such as `RequiredAttribute`, using `CodeAttributes` in C#, an ambiguous reference error is encountered in the generated class. This issue arises specifically when using the following line of code:

```csharp
col.Property.CustomAttributes.Add(new CodeAttribute(typeParser.Parse<RequiredAttribute>()));
```

### Encountered Error
The error message received is as follows:

```
Error CS0104: 'DataType' is an ambiguous reference between 'System.ComponentModel.DataAnnotations.DataType' and 'LinqToDB.DataType'
```

### Current Limitation
As of now, there doesn't appear to be a straightforward method to specify attributes using their fully qualified namespace within the `CodeAttributes` addition. This limitation poses a challenge in resolving the ambiguous reference.
